lib/ibazel_watcher.js

```diff
@@ -0,0 +1,27 @@
+const {createInterface} = require('readline');
+
+// ibazel will write this string after a successful build
+// We don't want to re-trigger tests if the compilation fails, so
+// we should only listen for this event.
+const IBAZEL_NOTIFY_BUILD_SUCCESS = 'IBAZEL_BUILD_COMPLETED SUCCESS';
+
+function watch(fileList) {
+  // ibazel communicates with us via stdin
+  const rl = createInterface({input: process.stdin, terminal: false});
+  rl.on('line', chunk => {
+    if (chunk === IBAZEL_NOTIFY_BUILD_SUCCESS) {
+      fileList.refresh();
+    }
+  });
+  rl.on('close', () => {
+    // Give ibazel 5s to kill our process, otherwise do it ourselves
+    setTimeout(() => {
```

Comment:
I think this is unnecessary. If the stream is closed it will be because you're being killed and it is a reasonable response to just quit immediately.